extensions/wikia/Recirculation/js/helpers/sponsoredContent.js

```diff
@@ -38,9 +39,28 @@ define('ext.wikia.recirculation.helpers.sponsoredContent', [
 	}
 
 	function getApplicableContent(sponsoredContent) {
-		return sponsoredContent.filter(function (el) {
-			return !el.geos.length || el.geos.indexOf(userGeo) !== -1;
+		var geoSpecificContent = sponsoredContent.filter(function (el) {
+			return (!el.geos.length || el.geos.indexOf(userGeo) !== -1) && (!el.wikiIds.length || el.wikiIds.indexOf(w.wgCityId) !== -1);
 		});
+		var siteSpecificContent = sponsoredContent.filter(function (item) {
+			return item.wikiIds.indexOf(w.wgCityId) !== -1;
+		});
+
+		var geoAndSiteSpecificContent = siteSpecificContent.filter(function (item) {
```

Comment:
I think this could be done without `some`, by filtering those items against `userGeo`